crates/client/mempool/src/l1.rs

```diff
@@ -12,6 +13,7 @@ pub struct GasPriceProvider {
     data_gas_price_sync_enabled: Arc<AtomicBool>,
     strk_gas_price_sync_enabled: Arc<AtomicBool>,
     strk_data_gas_price_sync_enabled: Arc<AtomicBool>,
+    pub oracle_provider: Option<Arc<Box<dyn Oracle>>>,
```

Comment:
```suggestion
    pub oracle_provider: Option<Arc<dyn Oracle>>,
```
Because Arc already provides heap allocation and dynamic dispatch for trait objects, making Box redundant here. This double indirection adds unnecessary complexity.
Need a specific reason to keep both Arc and Box?